pkg/framework/custom_junit_reporter.go

```diff
@@ -0,0 +1,248 @@
+			suite.Skipped += 1
+		case types.SpecStatePending:
+			test.Skipped = &JUnitSkipped{Message: "pending"}
+			suite.Disabled += 1
+		case types.SpecStateFailed:
+			test.Failure = &JUnitFailure{
+				Message:     spec.Failure.Message,
+				Type:        "failed",
+				Description: fmt.Sprintf("%s\n%s", spec.Failure.Location.String(), spec.Failure.Location.FullStackTrace),
+			}
+			suite.Failures += 1
+		case types.SpecStateInterrupted:
+			test.Error = &JUnitError{
+				Message:     "interrupted",
+				Type:        "interrupted",
+				Description: interruptDescriptionForUnstructuredReporters(spec.Failure),
+			}
+			suite.Errors += 1
+		case types.SpecStateAborted:
+			test.Failure = &JUnitFailure{
+				Message:     spec.Failure.Message,
+				Type:        "aborted",
+				Description: fmt.Sprintf("%s\n%s", spec.Failure.Location.String(), spec.Failure.Location.FullStackTrace),
+			}
+			suite.Errors += 1
+		case types.SpecStatePanicked:
+			test.Error = &JUnitError{
+				Message:     spec.Failure.ForwardedPanic,
+				Type:        "panicked",
+				Description: fmt.Sprintf("%s\n%s", spec.Failure.Location.String(), spec.Failure.Location.FullStackTrace),
+			}
+			suite.Errors += 1
+		}
+
+		suite.TestCases = append(suite.TestCases, test)
+	}
+
+	junitReport := JUnitTestSuites{
+		Tests:      suite.Tests,
+		Disabled:   suite.Disabled + suite.Skipped,
+		Errors:     suite.Errors,
+		Failures:   suite.Failures,
+		Time:       suite.Time,
+		TestSuites: []JUnitTestSuite{suite},
+	}
+
+	f, err := os.Create(dst)
+	if err != nil {
+		return err
+	}
+	_, err2 := f.WriteString(xml.Header)
+	if err2 != nil {
+		klog.Fatal(err2)
+	}
+	encoder := xml.NewEncoder(f)
+	encoder.Indent("  ", "    ")
+	err3 := encoder.Encode(junitReport)
+	if err3 != nil {
+		klog.Fatal(err3)
+	}
+	return f.Close()
+}
+
+// This function generates folder structure for the rp_preproc tool with logs for upload in Report Portal
+func GenerateRPPreprocReport(report types.Report) {
+	for i := range report.SpecReports {
+		reportSpec := report.SpecReports[i]
+		//generate folders only for failed tests
+		if !reportSpec.Failure.IsZero() {
+			if reportSpec.LeafNodeType == types.NodeTypeIt {
+				name := getClassnameFromReport(reportSpec) + "." + shortenStringAddHash(reportSpec)
+				filePath := "rp_preproc/attachments/xunit/" + name
+				if err := os.MkdirAll(filePath, os.ModePerm); err != nil {
+					klog.Fatal(err)
+				} else {
+					writeLogInFile(filePath+"/ginkgoWriter.log", reportSpec.CapturedGinkgoWriterOutput)
+					writeLogInFile(filePath+"/stdOutErr.log", reportSpec.CapturedStdOutErr)
+					writeLogInFile(filePath+"/failureMessage.log", reportSpec.FailureMessage())
+					writeLogInFile(filePath+"/failureLocation.log", reportSpec.FailureLocation().FullStackTrace)
+				}
+			}
+		}
+	}
+}
+
+func writeLogInFile(filePath string, log string) {
+	// Do not create empty files
+	if len(log) != 0 {
+		f, err := os.Create(filePath)
+		if err != nil {
+			klog.Fatal(err)
```

Comment:
should not be better to use `klog.Error(err)`. This will panick and exit the tests

Reply:
Thanks, I have changed it :)